frontend/src/components/utils/MultiUserSelector.tsx

```diff
@@ -0,0 +1,249 @@
+import React, { useContext, useEffect, useState } from "react";
+import { api, useBetterQuery } from "../../services/apiService";
+import Select from "react-select";
+import { ErrorMessage } from "./ErrorPage";
+import { Loading } from "./Loading";
+import { Col, Row } from "react-bootstrap";
+import { User } from "../../models/user";
+import { UserAvatarCard } from "./avatar/UserAvatarCard";
+import { Size } from "../../utils/size";
+import { UserContext } from "../../services/authService";
+import { DebounceInput } from "react-debounce-input";
+
+export const MultiUserSelector = ({ onChange }) => {
+    const loggedUser = useContext(UserContext);
+
+    const [searchValue, setSearchValue] = useState("");
+    const [promotion, setPromotion] = useState<any>(null);
+
+    const [userStatus, setUserStatus] = useState<
+        "loading" | "error" | "success"
+    >();
+    const [userError, setUserError] = useState<any>();
+
+    const [users, setUsers] = useState<{
+        [key: string]: { user: User; selected: boolean };
+    }>({});
+
+    const selectedUsers = Object.entries(users)
+        .filter(([_, { selected }]) => selected)
+        .map(([_, { user }]) => user)
+        .sort(compareUsers);
+
+    const unSelectedUsers = Object.entries(users)
+        .filter(([_, { selected }]) => !selected)
+        .map(([_, { user }]) => user)
+        .sort(compareUsers);
+
+    const {
+        data: promotions,
+        status: promotionsStatus,
+        error: promotionsError,
+    } = useBetterQuery<{
+        promotions: string[];
+    }>(["promotions.list"], api.users.listPromotions, {
+        refetchOnWindowFocus: false,
+    });
+
+    useEffect(() => {
+        setPromotion({
+            value: loggedUser?.promotion || "",
+            label: loggedUser?.promotion || "",
+        });
```

Comment:
Is there a problem if you put it line 17 in the state initialization?
It worked when I tried